jaxrs-api/src/main/java/javax/ws/rs/core/EntityTag.java

```diff
@@ -119,10 +116,7 @@ public boolean equals(final Object obj) {
      */
     @Override
     public int hashCode() {
-        int hash = 3;
-        hash = 17 * hash + (this.value != null ? this.value.hashCode() : 0);
-        hash = 17 * hash + (this.weak ? 1 : 0);
-        return hash;
+        return Objects.hash(this.value, this.weak);
```

Comment:
One should never mix bug fixes with enhancements in the same commit. It prevents committers from picking bug fixes into lts branches.

Reply:
ok, I understand the issue, but I think in this case this is not really a bug-fix, but only a simplification of the code (since no outside observable behavior is changed). So for me doing a PR "simplify hashcode/equals" instead of only removing one "if"-statement would be ok, but I can split the PR if you want me to.

Reply:
No need to split PRs. Just split commits within the same PR.